cardano-sync/src/Cardano/Sync/Types.hs

```diff
@@ -66,3 +67,14 @@ newtype BlockId = BlockId Int
 -- @Word64@ is valid as well.
 newtype MetaId = MetaId Int
     deriving (Eq, Show)
+
+-- The metrics we use.
+-- PLEASE do not put this into ENV, it's defeating the purpose
+-- of these constructs.
+data MetricsLayer = MetricsLayer
```

Comment:
Why is this a "Layer"?

Reply:
Like a `DataLayer`, we used names `Layer` to describe such interfaces. I think the wallet does a similar thing, it's not a first. I think the name fits quite well. It's a layer that can be cascaded (like the caching layer on top of a db layer), and it's an interface, a small surface that "touches" the world.

Reply:
The name `MetricsSetters` is a much better name. It removes the confusing "layer" from the name and "setters" describes *exactly* what the components of this struct do.

Yes, naming things in hard. It took me some time to come up with this improvement.